pvm/ts/src/rpc.ts

```
@@ -343,55 +341,39 @@ export class RPC {
-                        return makeErrorRPCResponse(id, `Deal operation failed: ${error.message || "Unknown error"}`);
+                case RPCMethods.PERFORM_ACTION: {
+                    const [from, to, action, amount, data] = request.params as RPCRequestParams[RPCMethods.PERFORM_ACTION];
+
+                    switch (action) {
+                        case "deal":
+                            try {
+                                const command = new DealCommand(to, from, validatorPrivateKey, data || "");
+                                result = await command.execute();
+
+                                // Check if the result indicates an error but wasn't thrown as an exception
+                                if (result && result.data && !result.data.success) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)